nemo/collections/nlp/modules/common/megatron/language_model.py

```diff
@@ -746,10 +746,7 @@ def forward(
                 ptuning_adapter = self.get_adapter_module(AdapterName.PTUNING_ADAPTER)
                 v = ptuning_adapter.virtual_tokens
                 if ptuning_adapter and _sq >= v:  # The sequence should be longer the v to insert virtual embeddings.
-                    strategy = ptuning_adapter.adapter_strategy
-                    virtual_embeddings = self.forward_single_enabled_adapter_(
-                        _bs, ptuning_adapter, adapter_name=AdapterName.PTUNING_ADAPTER, adapter_strategy=strategy,
-                    )
+                    virtual_embeddings = ptuning_adapter(_bs)
```

Comment:
Each adapter type has its own strategy - by removing this you're hard coding the logic and side stepping the strategy. Its not necessary to do that is it?

Reply:
Agreed, its not necessary, but just makes the readability and maintainability easier. Its just clearer what is happening to read the code and see a residual connection rather than follow up with where a default strategy is coming from.

Reply:
I think this is mostly bypassing the entire system. I would suggest reverting this so that the strategy kicks in and to only update the adapter strategy defaults in all adapters.

Reply:
oh one other limitation i think is that i can not pass additional args? for example for ptuning_adapter the forward now accepts an additional arg like `used_cached_reps`

Reply:
After discussion, seems this is never going to be needed by nlp domain, so its fine to directly call adaper